exercises/practice/rtos-static-scheduling/.docs/instructions.md

```diff
@@ -0,0 +1,54 @@
+# Instructions
+
+This problem is modeled after real-time operating system nonpreemptive static 
+scheduling, but is also similar to the Job-Shop Scheduling Problem (JSP) with 
+only one resource.  This problem assumes that all tasks are known statically.  In 
+other words, the release times of all tasks in this problem are known at the beginning 
+of the program.  In practice, the release time of tasks are not known at the start of a 
+program, and may be generated at any time through I/O, interrupts, etc.
+
+Given a list of tasks with parameters, determine a schedule in which
+the tasks can execute sequentially with no overlap.  Since the problem is modeled 
+after RTOS process scheduling, we will assume there is a nonzero time required between 
+tasks, also known as the context-switching overhead. There are two different types 
+of tasks: periodic and dynamic.  Both types have a task ID parameter which is a
+string used to differentiate between the different tasks in the task list.
+
+A **periodic task** is defined as a task that executes periodically.  In other words,
+the task will execute once every T seconds where T is defined as the period.
+A periodic task is defined by the following parameters:
+* Start Time - time the first repetition, or instance, of the task starts
+* Period - time between the start of the task repetitions
+* Execution Time - time each repetition runs for 
+* End Time - time the entire task stops getting scheduled
+
+A **dynamic task** is defined as a task that only executes once.
+A dynamic task is defined by the following parameters:
+* Release Time - first time the task CAN be scheduled (NOT the same as start time)
+* Execution Time - time the task runs for 
+* Deadline - time in which the task must finish execution by
+
+There are two inputs to the function to be implemented.  The first input is a list of
+instances of `RealTimeTask`.  Instances of `RealTimeTask` will either be input as
+an instance of `PeriodicRealTimeTask` or `DynamicRealTimeTask` with all task parameters as integers.  
+The `isinstance()` built-in Python function can be used to differentiate between the two tasks 
+in the list.  The order of the list is irrelevant; look only at the parameters of the tasks.  
+The second input is a single integer that specifies the context switching overhead.  The context
+switching overhead will always be nonzero.
```

Comment:
Should it always be positive nonzero, or just nonzero?

Reply:
Positive nonzero.  But it doesn't make any sense for it to be negative nonzero, so I don't see the point in specifying positive nonzero.